src/qibo/backends/numpy.py

```diff
@@ -467,18 +473,31 @@ def execute_circuit_repeated(self, circuit, initial_state=None, nshots=None):
 
             if circuit.measurements:
                 result = CircuitResult(self, circuit, state, 1)
-                results.append(result.samples()[0])
+                sample = result.samples()[0]
+                results.append(sample)
+                if not circuit.density_matrix:
+                    probabilities += result.probabilities()
+                    frequencies.append(sample[0])
```

Comment:
Two remarks:
1. These are not frequencies yet, they are samples, right?
2. I think the indexing `sample[0]` is not working for 2 qubits and more. See my comment.